ssingh95808/Media-Insight
# Use case 3 text summarization using abstractive model

# 📝 Add Text Summarization Feature
## Changes
Implemented usecase3: Added text summarization feature using abstractive BART model, and made integration for our system including frontend interface and backend API.

## Known Issues
- The React UI needs to be fixed.

## Type
- [ ] 🎨 Code Style/Structure

## Details


### Implementation
1. Added text input field and submit button to frontend
2. Implemented BART-based text summarization API
3. Added async task processing
4. Implemented polling for results

### Testing

Test Steps:
1. Start development server
2. Input long text and submit
3. Verify summary generation

## 📸 Screenshots
<details>
<summary>UI Interface</summary>
<img width="1092" alt="image" src="https://github.com/user-attachments/assets/fb18ecc0-652b-463c-bc15-873ff8281000">
</details>

## ✅ Checklist
- [ ] Added unit tests for summarization logic
- [ ] All CI tests passing
